client/dashboard/sites/reset-site-modal/index.tsx

```diff
@@ -0,0 +1,241 @@
+} ) {
+	const { mutate } = useMutation< void, ErrorResponse >( resetSiteMutation( site.ID ) );
+	const [ isMutatePending, setIsMutatePending ] = useState( false );
+	const [ textValue, setTextValue ] = useState< string >( '' );
+
+	const handleReset = () => {
+		if ( isMutatePending ) {
+			return;
+		}
+
+		setIsMutatePending( true );
+		mutate( undefined, {
+			onSuccess: () => {
+				setIsMutatePending( false );
+				onSuccess();
+			},
+			onError: ( error: ErrorResponse ) => {
+				setIsMutatePending( false );
+
+				if ( error.code === 'staging_sites_not_supported' ) {
+					onError( error.message );
+				} else {
+					onError( __( 'The site could not be reset due to an error. Please try again.' ) );
+				}
+			},
+		} );
+	};
+
+	return (
+		<VStack spacing={ 6 }>
+			<ExportNotice siteSlug={ site.slug } />
+			<Text as="p">
+				{ sprintf(
+					/* translators: %s: site domain */
+					__(
+						"Resetting %s will remove all of its content but keep the site and its URL up and running. You'll also lose any modifications you've made to your current theme. This cannot be undone."
+					),
+					siteDomain
+				) }
+			</Text>
+			<ContentInfo siteContent={ siteContent } siteDomain={ siteDomain } />
+			<TextControl
+				__nextHasNoMarginBottom
+				__next40pxDefaultSize
+				label={ __( 'Type the site domain to confirm' ) }
+				value={ textValue }
+				onChange={ ( value ) => setTextValue( value.trim() ) }
+				help={ sprintf(
+					/* translators: %s: site domain */
+					__( 'The site domain is: %s' ),
+					siteDomain
+				) }
+			/>
+			<HStack spacing={ 4 } justify="flex-end">
+				<Button variant="link" onClick={ onClose }>
+					{ __( 'Cancel' ) }
+				</Button>
+				<Button
+					variant="primary"
+					disabled={ textValue !== siteDomain }
+					isBusy={ isMutatePending }
+					onClick={ handleReset }
+				>
+					{ __( 'Reset site' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+export default function ResetSiteModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const {
+		data: resetStatus,
+		isFetching: isFetchingResetStatus,
+		isLoading: isLoadingResetStatus,
+		refetch: refetchResetStatus,
+	} = useQuery< SiteResetStatus >( {
+		...siteResetStatusQuery( site.ID ),
+	} );
+	const { data: siteContent, isLoading: isLoadingSiteContent } = useQuery< SiteContent >(
+		siteResetContentSummaryQuery( site.ID )
+	);
+	const queryClient = useQueryClient();
+
+	const siteDomain = site?.URL ? new URL( site.URL ).hostname : '';
+
+	const showSuccessNotice = useCallback( () => {
+		createSuccessNotice(
+			sprintf(
+				/* translators: %s: site domain */
+				__( '%s has been reset.' ),
+				siteDomain
+			),
+			{ type: 'snackbar' }
+		);
+	}, [ createSuccessNotice, siteDomain ] );
+
+	useEffect( () => {
+		if ( resetStatus?.status === 'completed' && ! isFetchingResetStatus ) {
```

Comment:
I don't know if this is right. 

What was happening is that everything works properly. The request resolves to `completed` and we close the window, but if the user clicks to open the dialog again, it seems like `status = 'completed'` was still being served from the cache, regardless of whether I set the stale time or invalidated all the queries. 

Because of that, it would immediately close again which seems to hard reset the cache properly.

Reply:
Ah I see. I think this is related to persisting data to localStorage.